unit_testing/formats.js

```diff
@@ -0,0 +1,41 @@
+const PropTypes = require('prop-types');
+const exact = require('prop-types-exact');
+
+const link = module.exports.link = exact({
+  title: PropTypes.string.isRequired,
+  author: PropTypes.oneOfType([
+    PropTypes.shape(exact({
+      name: PropTypes.string.isRequired,
+      url: PropTypes.string,
+    })),
+    PropTypes.string,
+  ]),
+  time: PropTypes.string,
+  url: PropTypes.string,
+  video_id: PropTypes.string,
+  playlist_id: PropTypes.string,
+  source: PropTypes.string,
+})
+
+const video = module.exports.video = exact({
+  title: PropTypes.string.isRequired,
+  video_number: PropTypes.number.isRequired,
+  date: PropTypes.instanceOf(Date).isRequired,
+  video_id: PropTypes.string.isRequired,
+  repository: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.oneOf([false]),
```

Comment:
Can `repository` be false? I thought it shouldn't be, but we are missing a repository for Coding Challenge 19 and its file has a false here.